src/Widgets/Views/MainView.vala

```diff
@@ -91,8 +91,9 @@ public class Bluetooth.Widgets.MainView : Gtk.Box {
 
         object_manager.device_removed.connect ((device) => {
             devices_box.get_children ().foreach ((child) => {
-                if (child is Bluetooth.Widgets.Device) {
-                    ((Bluetooth.Widgets.Device) child).destroy ();
+                var device_child = child as Bluetooth.Widgets.Device;
+                if (device_child != null && device_child.device.modalias == device.modalias) {
```

Comment:
Again, could we have a `compare` method inside the `Device` class like that?

```
public bool compare (Device device) {
    return modalias == device.modalias;
}
```

It would just make this easier to compare devices in the future.